libraries/AP_BattMonitor/AP_BattMonitor_DJI.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_Backend.h"
+
+class AP_BattMonitor_DJI : public AP_BattMonitor_Backend
+{
+public:
+
+    /// Constructor
+    AP_BattMonitor_DJI(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params);
+
+    /// Read the battery voltage and percentage.
+    void read() override;
+
+    /// returns true if battery monitor provides consumed energy info
+    bool has_consumed_energy() const override { return true; }
+
+    /// returns true if battery monitor provides current info
+    bool has_current() const override { return true; }
```

Comment:
battery is not reporting any current info / or I need to spend more time to reverse engineering, may that will add in future.

Reply:
Okay, in that case then this should be returning false as we don't have it. Returning true here would cause a number of libraries to use 0 amps for scaling their controllers which would be bad. Similarly `has_consumed_energy()` should return false.

If you look at the implementation of `capacity_remaining_pct` it just uses the capacities directly, and doesn't require either of these methods to be returning true.